index.js

```diff
@@ -141,44 +176,66 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+let ticketNumber = 0;
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+function takeATicketNumber(arr) {
+  ticketNumber++;
+  line.push(ticketNumber)
+  return `Welcome. You are ticket number ${ticketNumber}`
+}
+
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+
+function ticketNumberGeneratorFunc() {
+  let counter = [];
+  function newTakeATicketNumber(arr) {
+    if (!arr.length) {
+      arr.push(1);
+      counter.push(1);
+    } else {
+      arr.push(parseInt(`${arr.length + 1}`));
+      counter.push(parseInt(`${arr.length + 1}`));
+    }
+    return `Welcome. You are ticket number ${counter.length}`;
+  }
+  return newTakeATicketNumber
+}
```

Comment:
This function works but can be significantly cleaned up by using a clue from the readme. "Write a higher order function called ticketNumberGeneratorFunc that returns a nested takeATicketNumber function. The other functionality should remain the same."
In other words we just need to use the function from the previous question inside of the new ticketNumberGeneratorFunce and return it

function ticketNumberGeneratorFunc() {
  let ticket = 0

  return function takeATicketNumber(line) {
    ticket++
    line.push(ticket)
    return `Welcome, you are ticket number ${ticket}`
  }
}